addons/mrp/wizard/mrp_batch_produce.py

```diff
@@ -85,10 +87,21 @@ def _production_text_to_object(self, mark_done=False):
             components_list.append(components_line)
 
         productions = self.production_id._split_productions({self.production_id: productions_amount})
-        lots = self.env['stock.lot'].create([{
-            'name': lot_name,
-            'product_id': productions.product_id.id
-        } for lot_name in productions_lot_list])
+
+        lots = self.env['stock.lot'].search(
+            domain=[
+                ('name', '=', self.lot_name)
+            ],
+            limit=1,
+        )
+        was_first_sn_created = any(lots)
+        for lot_name in productions_lot_list:
+            if lot_name == self.lot_name and was_first_sn_created:
+                continue
+            lots = lots + self.env['stock.lot'].create({
+                'name': lot_name,
+                'product_id': productions.product_id.id
+            })
```

Comment:
This change seems unrelated to the imp you're adding in. I think I see what you are trying to do though. This change should be moved to another commit message + needs to be improved because any of the lot names in `productions_lot_list` could already exist. I'll come discuss about what should the behavior be here.